Tests/DatadogTests/Datadog/URLSessionAutoInstrumentation/Interception/URLSessionInterceptorTests.swift

```diff
@@ -101,14 +94,77 @@ class URLSessionInterceptorTests: XCTestCase {
+        XCTAssertTrue(
+            interceptor.injectTracingHeadersToFirstPartyRequests,
+            "Tracing headers should be injected when both Tracing and RUM instrumentations are enabled."
+        )
+        XCTAssertEqual(
+            interceptor.additionalHeadersForFirstPartyRequests,
+            [TracingHTTPHeaders.originField: TracingHTTPHeaders.rumOriginValue],
+            "Additional `x-datadog-origin: rum` header should be injected when both Tracing and RUM instrumentations are enabled."
+        )
     }
 
     // MARK: - URLRequest Interception
 
-    func testGivenTracerRegistered_whenInterceptingRequests_itInjectsSpanContextOnlyToFirstPartyRequests() throws {
+    private func mockConfiguration(
+        tracingInstrumentationEnabled: Bool,
+        rumInstrumentationEnabled: Bool
+    ) -> FeaturesConfiguration.URLSessionAutoInstrumentation {
+        return .mockWith(
+            userDefinedFirstPartyHosts: ["first-party.com"],
+            sdkInternalURLs: ["https://dd.internal.com"],
+            instrumentTracing: tracingInstrumentationEnabled,
+            instrumentRUM: rumInstrumentationEnabled
+        )
+    }
+
+    func testGivenTracingAndRUMInstrumentationEnabled_whenInterceptingRequests_itInjectsTracingContextToFirstPartyRequests() throws {
+        // Given
+        let interceptor = URLSessionInterceptor(
+            configuration: mockConfiguration(tracingInstrumentationEnabled: true, rumInstrumentationEnabled: true),
+            handler: handler
+        )
+        Global.sharedTracer = Tracer.mockAny()
+        defer { Global.sharedTracer = DDNoopGlobals.tracer }
+
+        // When
+        let interceptedFirstPartyRequest = interceptor.modify(request: firstPartyRequest)
+        let interceptedThirdPartyRequest = interceptor.modify(request: thirdPartyRequest)
+        let interceptedInternalRequest = interceptor.modify(request: internalRequest)
+
+        // Then
+        XCTAssertNotNil(interceptedFirstPartyRequest.allHTTPHeaderFields?[TracingHTTPHeaders.traceIDField])
+        XCTAssertNotNil(interceptedFirstPartyRequest.allHTTPHeaderFields?[TracingHTTPHeaders.parentSpanIDField])
+        XCTAssertEqual(interceptedFirstPartyRequest.allHTTPHeaderFields?[TracingHTTPHeaders.originField], TracingHTTPHeaders.rumOriginValue)
+        XCTAssertNil(interceptedThirdPartyRequest.allHTTPHeaderFields?[TracingHTTPHeaders.traceIDField])
+        XCTAssertNil(interceptedThirdPartyRequest.allHTTPHeaderFields?[TracingHTTPHeaders.parentSpanIDField])
+        XCTAssertNil(interceptedThirdPartyRequest.allHTTPHeaderFields?[TracingHTTPHeaders.originField])
+        XCTAssertNil(interceptedInternalRequest.allHTTPHeaderFields?[TracingHTTPHeaders.traceIDField])
+        XCTAssertNil(interceptedInternalRequest.allHTTPHeaderFields?[TracingHTTPHeaders.parentSpanIDField])
+        XCTAssertNil(interceptedInternalRequest.allHTTPHeaderFields?[TracingHTTPHeaders.originField])
```

Comment:
i guess we expect `request.allHTTPHeaderFields == nil/unchanged` in those 2 cases:
```swift
XCTAssertNil(interceptedThirdPartyRequest.allHTTPHeaderFields?)
XCTAssertNil(interceptedInternalRequest.allHTTPHeaderFields?)
```
?
and the next lines might already cover them:
```swift
XCTAssertEqual(thirdPartyRequest, interceptedThirdPartyRequest)
XCTAssertEqual(internalRequest, interceptedInternalRequest)
```
if `HTTPHeaderFields` are changed, those assertions fail; don't they?

Reply:
Yes, inspecting `.allHTTPHeaderFields` is redundant if we assert the full request equality later 👍. However, due to request equality bug in iOS 14 SDK: https://openradar.appspot.com/radar?id=4988276943355904 it seemed safer to do it before. To be more verbose on this, I added extra helper: `assertRequestsEqual(_:_:)` which implements the workaround. Redundant checks are now removed 👌.